src/couch/src/couch_ejson_size.erl

```diff
@@ -53,3 +62,130 @@ ejson_size(false) ->
 
 ejson_size(Atom) when is_atom(Atom) ->
     ejson_size(atom_to_binary(Atom, utf8)).
+
+
+utf8_string_size(Binary) ->
+    %% 2 is for open and closing "
+    utf8_string_size(Binary, 2).
+
+utf8_string_size(<<0:1, C:7/integer, Rest/binary>>, Acc) ->
+    case C of
+        ?quotation_mark -> utf8_string_size(Rest, Acc + 2);
+        ?reverse_solidus -> utf8_string_size(Rest, Acc + 2);
+        ?backspace -> utf8_string_size(Rest, Acc + 2);
+        ?horizontal_tab -> utf8_string_size(Rest, Acc + 2);
+        ?newline -> utf8_string_size(Rest, Acc + 2);
+        ?formfeed -> utf8_string_size(Rest, Acc + 2);
+        ?carriage_return -> utf8_string_size(Rest, Acc + 2);
+        NeedEscape when NeedEscape =< 16#001F -> utf8_string_size(Rest, Acc + 6);
+        _Else -> utf8_string_size(Rest, Acc + 1)
+    end;
+%% 10000 .. 10000007F
+utf8_string_size(<<2#1111:4, _:28/bitstring, Rest/binary>>, Acc) ->
+    utf8_string_size(Rest, Acc + 4);
+%% 0800 .. FFFF
+utf8_string_size(<<2#111:3, _:21/bitstring, Rest/binary>>, Acc) ->
+    utf8_string_size(Rest, Acc + 3);
+%% 0080 .. 07ff
+utf8_string_size(<<2#11:2, _:14/bitstring, Rest/binary>>, Acc) ->
+    utf8_string_size(Rest, Acc + 2);
+utf8_string_size(<<>>, Acc) ->
+    Acc.
+
+
+-ifdef(TEST).
+
+-include_lib("eunit/include/eunit.hrl").
+
+proper_test_() ->
+    PropErOpts = [
+        {to_file, user},
+        {max_size, 5},
+        {numtests, 1000}
+    ],
+    properties_tests(properties(), 30, PropErOpts).
+
+properties() ->
```

Comment:
I wanted to remove this function and just hardcode the list of properties. I forgot to do it.